### PR TITLE
Terraform resources

### PR DESCRIPTION
**Pull request by the-smooth-operator**

### CODE REVIEW

**Comment by the-smooth-operator** [comment]
@limed can you please review this code? Please point out everything you find you don't like :)

File: terraform/docker-registry.tf [hunk 1]
```diff
@@ -0,0 +1,5 @@
+resource "aws_ecr_repository" "discourse" {
+  name = "discourse-${terraform.workspace}"
+  tags                        = "${merge(var.common-tags, var.workspace-tags)}"
```
Comment (limed) [inline terraform/docker-registry.tf]: Nitpick, formatting here. I recommend running this through `terraform fmt`

File: terraform/redis.tf [hunk 2]
```diff
@@ -0,0 +1,40 @@
+resource "aws_elasticache_cluster" "discourse" {
+  cluster_id           = "discourse-${terraform.workspace}-redis"
+  engine               = "redis"
+  node_type            = "${var.redis-instance}"
+  num_cache_nodes      = "${var.redis-num-nodes}"
+  engine_version       = "${var.redis-version}"
+  parameter_group_name = "default.redis${var.redis-version}"
+	subnet_group_name    = "${aws_db_subnet_group.discourse-redis.id}"
```
Comment (limed) [inline terraform/redis.tf]: Nitpick formatting, run through `terraform fmt`

File: terraform/variables.tf [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+
+variable "account" {
+  default = ""
+}
+
+#########
+# Redis #
+#########
+
+variable "redis-instance" {
```
Comment (limed) [inline terraform/variables.tf:9]: I realize you are setting the actual values in you `tfvars` file but I recommend that you also set some sort of default value here as well.

File: terraform/variables.tf [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+variable "redis-version" {
+  default = "5.0.4"
+}
+
+##########
+#  PSQL  #
+##########
+
+variable "psql-instance" {
+  default = ""
```
Comment (limed) [inline terraform/variables.tf]: Same as above, recommend you set a default value here too and no leave it to `""`

File: terraform/docker-registry.tf (hunk 1 above)
Comment (the-smooth-operator) [reply terraform/docker-registry.tf]: :+1: I should add a pre commit hook doing this

File: terraform/variables.tf (hunk 3 above)
Comment (the-smooth-operator) [reply terraform/variables.tf:9]: :+1: I should add a pre commit hook doing this

File: terraform/variables.tf (hunk 4 above)
Comment (the-smooth-operator) [reply terraform/variables.tf]: thanks, that's useful input

**Pull request merged**